src/test/regress/expected/shard_rebalancer.out

```diff
@@ -2378,3 +2378,63 @@ SELECT rebalance_table_shards();
 
 CALL citus_cleanup_orphaned_shards();
 DROP TABLE test_rebalance_with_index CASCADE;
+-- Test rebalancer with disabled worker
+SET citus.shard_replication_factor TO 2;
+DROP TABLE IF EXISTS test_rebalance_with_disabled_worker;
+CREATE TABLE test_rebalance_with_disabled_worker (a int);
+SELECT create_distributed_table('test_rebalance_with_disabled_worker', 'a', colocate_with:='none');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_disable_node('localhost', :worker_2_port);
+ citus_disable_node
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT public.wait_until_metadata_sync(30000);
+ wait_until_metadata_sync
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT rebalance_table_shards('test_rebalance_with_disabled_worker');
+ rebalance_table_shards
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT citus_activate_node('localhost', :worker_2_port);
+ citus_activate_node
+---------------------------------------------------------------------
+                  35
+(1 row)
+
+DROP TABLE test_rebalance_with_disabled_worker;
+-- Test rebalance with all shards excluded
+DROP TABLE IF EXISTS test_with_all_shards_excluded;
+CREATE TABLE test_with_all_shards_excluded(a int);
+SELECT create_distributed_table('test_with_all_shards_excluded', 'a', colocate_with:='none', shard_count:=4);
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+SELECT shardid FROM pg_dist_shard;
+ shardid
+---------------------------------------------------------------------
```

Comment:
These shardIds might change if say we change the order of the tests.

To avoid such cases, we use the following trick to fix the shardIds:
```SQL
SET citus.next_shard_id TO 433000;
```

In this test file, we already have this, but as we re-connect to the coordinator via `\c - - - :master_port`, we lose the above SET.

So, we should re-set before create_distributed_table to something like the following, where we allow ~500 shards in this file before this line, which should be good enough:
```SQL
SET citus.next_shard_id TO 433500;
```

So that the shards have consistent shardIds.

(alternatively, we may NOT print the shardids and use some psql variable like `SELECT .. INTO variable FROM ...`), and then use the variable in `excluded_shard_list`. Though, still setting the `citus.next_shard_id` is a good practice